tests/lib/rules/no-loop-func.js

```diff
@@ -156,6 +156,58 @@ ruleTester.run("no-loop-func", rule, {
         {
             code: "for (let i = 0; i < 10; ++i) { for (let x in xs.filter(x => x != undeclared)) {  } }",
             languageOptions: { ecmaVersion: 6 }
+        },
+
+        // IIFE
+        {
+            code: `
+            let current = getStart();
+            while (current) {
+            (() => {
+                current;
+                current.a;
+                current.b;
+                current.c;
+                current.d;
+            })();
+            
+            current = current.upper;
+            }
+            `,
+            languageOptions: { ecmaVersion: 6 }
+        },
+        "for (var i=0; (function() { i; })(), i<l; i++) { }",
+        "for (var i=0; i<l; (function() { i; })(), i++) { }",
+        {
+            code: "for (let i = 0; i < 10; ++i) { (()=>{ i;})() }",
+            languageOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "for (let i = 0; i < 10; ++i) { (function a(){i;})() }",
+            languageOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: `
+            var arr = [];
+
+            for (var i = 0; i < 5; i++) {
+                arr.push((f => f)((() => i)()));
+            }
+            `,
+            languageOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: `
+            var arr = [];
+
+            for (var i = 0; i < 5; i++) {
+                arr.push((() => {
+                    return (() => i)();
+                })());
+            }
+            `,
+            languageOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unsafeRefs", data: { varNames: "'i'" }, type: "ArrowFunctionExpression" }]
```

Comment:
```suggestion
            languageOptions: { ecmaVersion: 6 }
```

This is in the `valid` array.